pkg/proto/protobuf_converters.go

```
@@ -1714,6 +1716,59 @@ func (c *ProtobufConverter) Block(block *g.Block) (Block, error) {
 	}, nil
 }
 
+func (c *ProtobufConverter) EndorseBlock(endorsement *g.EndorseBlock) (EndorseBlock, error) {
+	if endorsement == nil {
+		return EndorseBlock{}, errors.New("empty endorsement")
+	}
+	finalizedBlockID, err := NewBlockIDFromBytes(endorsement.FinalizedBlockId)
+	if err != nil {
+		return EndorseBlock{}, fmt.Errorf("failed to parse finalized block ID: %w", err)
+	}
+	endorsedBlockID, err := NewBlockIDFromBytes(endorsement.EndorsedBlockId)
+	if err != nil {
+		return EndorseBlock{}, fmt.Errorf("failed to parse endorsed block ID: %w", err)
+	}
+	return EndorseBlock{
+		EndorserIndex:        endorsement.EndorserIndex,
+		FinalizedBlockID:     finalizedBlockID,
+		FinalizedBlockHeight: endorsement.FinalizedBlockHeight,
+		EndorsedBlockID:      endorsedBlockID,
+		Signature:            endorsement.Signature,
+	}, nil
+}
+
+func (c *ProtobufConverter) FinalizationVoting(finalizationVoting *g.FinalizationVoting) (FinalizationVoting, error) {
+	if finalizationVoting == nil {
+		return FinalizationVoting{}, errors.New("empty finalization voting")
+	}
+	conflictEndorsements := make([]EndorseBlock, 0, len(finalizationVoting.ConflictEndorsements))
+	for i, ce := range finalizationVoting.ConflictEndorsements {
+		if ce == nil {
+			continue
+		}
```

Maybe return an error in this case? Or it's expected and possible situation?